inputstream.py

```diff
@@ -0,0 +1,25 @@
+from operator import methodcaller
+
+from antlr4 import InputStream
+
+
+class CaseTransformInputStream(InputStream):
```

Comment:
Can you describe quickly what this does? A lot of parts of SQL can be case sensitive. AFAIK DataCamp is doing case insensitive column names, but will this also transform SQL strings? (Maybe not too bad if it does)

Reply:
I guess even if it does render strings in upper case, if the AST process has access to the raw code it can set strings  back (I'm guessing converting to upper case is for speed of parsing?)

Reply:
It does the transform only for lexing, which is case sensitive.

Reply:
The current grammar has a lexer that supports both uppercase and lowercase.
I wrote this to test a different grammar that requires the transform and decided to include it for when we need it later.
[ANTLR documentation has more information](https://github.com/antlr/antlr4/blob/master/doc/case-insensitive-lexing.md).